drivers/bluetooth/hci/userchan.c

```diff
@@ -188,7 +188,7 @@ static void rx_thread(void *p1, void *p2, void *p3)
 		struct net_buf *buf;
 		size_t buf_tailroom;
 		size_t buf_add_len;
-		ssize_t len;
+		k_ssize_t len;
```

Comment:
Is this really correct? This driver is for native_sim, and some 10 lines later the return value from the host libc's `read()` is assigned to it.

Reply:
The definitions should be identical.

Reply:
I don't doubt that. I was mainly wondering where a `native_sim` driver like this conceptually sits in the software stack. It's calling the host OS APIs (like `read()`) so in that sense it could be seen to be above POSIX (in which case `ssize_t` would be more appropriate), however the driver also accesses Zephyr kernel APIs. Anyway, this isn't necessarily that critical (as you say, the types should be the same), which is why I didn't want to block this PR by requesting changes, but I do think it would be good to understand where `native_sim` driver code sits in the software stack.

Reply:
Yea, it's kind of a recursion. I would still say those drivers sit below (Zephyr's) POSIX API.

The namespace issue is kind of an unfortunate side-effect (not without solutions, ofc)